src/lib/UtapiClient.js

```diff
@@ -256,22 +256,32 @@ export default class UtapiClient {
     * @param {string} reqUid - Request Unique Identifier
     * @param {string} bucket - bucket name
     * @param {number} objectSize - size of object in bytes
+    * @param {number} prevObjectSize - previous size of object in bytes if this
+    * action overwrote an existing object
     * @param {callback} [cb] - (optional) callback to call
     * @return {undefined}
     */
-    pushMetricUploadPart(reqUid, bucket, objectSize, cb) {
+    pushMetricUploadPart(reqUid, bucket, objectSize, prevObjectSize, cb) {
         const callback = cb || this._noop;
         if (this.disableClient) {
             return callback();
         }
         const log = this.log.newRequestLoggerFromSerializedUids(reqUid);
         const timestamp = UtapiClient.getNormalizedTimestamp();
+
+        let oldObjSize = parseInt(prevObjectSize, 10);
+        oldObjSize = isNaN(oldObjSize) ? 0 : oldObjSize;
+        let newObjSize = parseInt(objectSize, 10);
+        newObjSize = isNaN(newObjSize) ? 0 : newObjSize;
```

Comment:
In both these situations, what are the contexts when parseInt could return NaN? Shouldn't we be logging something then (if a wrong value was passed, there is probably some interesting context associated with it?), and maybe stop function execution here rather than assume a size of zero and go along?